### PR TITLE
check for DX_AUTH_TOKEN with every dxpy api call

### PR DESCRIPTION
**Pull request by anujkumar93**
Currently DX_AUTH_TOKEN is picked up by dxpy only once upon importing dx.py module. Change this to be updated everytime a dxpy api is invoked.
Hence, shifted the check for DX_AUTH_TOKEN from settings to _wrap_dx_calls.

### CODE REVIEW

File: stor/dx.py [hunk 1]
```diff
@@ -82,9 +75,15 @@ def _dx_error_to_descriptive_exception(client_exception):
 
 
 @contextmanager
-def _propagate_dx_exceptions():
+def _wrap_dx_calls():
     """Bubbles all dxpy exceptions as `DNAnexusError` classes
     """
+    auth_token = settings.get()['dx']['auth_token']
+    if auth_token:  # pragma: no cover
```
Comment (jtratner) [inline stor/dx.py:83]: presumably this code ought to be covered.  pretty easy to use this with no-op credentials

File: stor/dx.py [hunk 2]
```diff
@@ -82,9 +75,15 @@ def _dx_error_to_descriptive_exception(client_exception):
 
 
 @contextmanager
-def _propagate_dx_exceptions():
+def _wrap_dx_calls():
     """Bubbles all dxpy exceptions as `DNAnexusError` classes
```
Comment (jtratner) [inline stor/dx.py]: docstring out of date

**Review by jtratner — APPROVED** [review]
fix docstring to note about setting auth token and good to merge

File: stor/dx.py (hunk 2 above)
Comment (anujkumar93) [reply stor/dx.py]: Fixed

**Pull request merged**